api/src/main/java/org/pmiops/workbench/access/AccessTierServiceImpl.java

```diff
@@ -75,27 +75,26 @@ public void addUserToAllTiers(DbUser user) {
   public void addUserToTier(DbUser user, DbAccessTier accessTier) {
     addToAuthDomainIdempotent(user, accessTier);
 
-    Optional<DbUserAccessTier> existingEntryMaybe =
-        userAccessTierDao.getByUserAndAccessTier(user, accessTier);
-
-    if (existingEntryMaybe.isPresent()) {
-      final DbUserAccessTier entryToUpdate = existingEntryMaybe.get();
-
-      // don't update if already ENABLED
-      if (entryToUpdate.getTierAccessStatusEnum() == TierAccessStatus.DISABLED) {
-        userAccessTierDao.save(
-            entryToUpdate.setTierAccessStatus(TierAccessStatus.ENABLED).setLastUpdated(now()));
-      }
-    } else {
-      final DbUserAccessTier entryToInsert =
-          new DbUserAccessTier()
-              .setUser(user)
-              .setAccessTier(accessTier)
-              .setTierAccessStatus(TierAccessStatus.ENABLED)
-              .setFirstEnabled(now())
-              .setLastUpdated(now());
-      userAccessTierDao.save(entryToInsert);
-    }
+    userAccessTierDao
```

Comment:
is this better?

Reply:
Yes, it is easier to follow the flow with this structure.